rename_imports.py

```diff
@@ -0,0 +1,78 @@
+
+from LSP.plugin.core.file_watcher import FileWatcherEvent, FileWatcherProtocol, get_file_watcher_implementation, FileWatcher
+from LSP.plugin.core.typing import  List
+import os
+import sublime
+
+
+def register_rename_import_file_watcher(root_path: str):
+    file_watcher = get_file_watcher_implementation()
+    if file_watcher:
+        watcher = file_watcher.create(
+            root_path=root_path,
+            patterns=["**/*.ts", "**/*.tsx", "**/*.js", "**/*.jsx"],
+            events=["create", "delete"],
+            ignores=['**/node_modules/**'],
+            handler=HandleRenameImport
+        )
+
+
+queued_events = []  # type: List[FileWatcherEvent]
+
+
+class HandleRenameImport(FileWatcherProtocol):
+    def on_file_event_async(events: List[FileWatcherEvent]) -> None:
+        global queued_events
+        queued_events.append(events[0])
+        sublime.set_timeout(check_if_there_was_a_rename, 10)
+
+
+def check_if_there_was_a_rename():
+    global queued_events
+    if len(queued_events) == 2:
+        first_event, new_name = queued_events[0]
+        second_event, old_name = queued_events[1]
+        if (first_event == 'create' and second_event == 'delete'):
+            # Rename detected
+            prompt_rename(old_name, new_name)
+    queued_events = []
+
+
+def prompt_rename(old_name: str, new_name: str) -> None:
+    settings = sublime.load_settings("LSP-typescript.sublime-settings")
+    update_imports_on_file_move_setting = settings.get('settings', {}).get('updateImportsOnFileMove', "prompt")
+    if update_imports_on_file_move_setting == "never":
+        return
```

Comment:
This could be checked before we even set up the watcher. Why set it up otherwise?

Reply:
Changed the logic so that the watcher wont be created if the `update_imports_on_file_move_setting` is `never`